design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,266 @@
+When removing an OSD (or OSDs) via the Rook Orchestrator Module, the Ceph mgr running in Kubernetes
+will also perform whatever Ceph operations are necessary to remove the OSD as would be expected by
+the user. This will replace the manual, imperative steps currently required by Rook-Ceph users to
+remove OSDs from the storage cluster documented
+[here for Rook v1.2](https://rook.io/docs/rook/v1.2/ceph-osd-mgmt.html#remove-an-osd).
+
+There is currently a proposal for a shared Ceph Mgr Orchestrator Module implementing common OSD
+removal steps [[link]](https://github.com/ceph/ceph/pull/32677) usable by any other Ceph Mgr
+Orchestrator module. The OSD removal mechanism should use this where possible.
+
+Rook will not automatically remove Kubernetes Deployments for OSDs unless
+`removeOSDsIfOutAndSafeToRemove: true`. This should be documented for users, and users should be
+advised to set this value to `true`.
+
+The `CephCluster` CRD allows users to add disks to nodes in several places including disks on all
+nodes in the Kubernetes cluster, to a specific node, or via disk pattern matching (in two different
+ways). To avoid the complexity of having to manage the logic for determining how to remove a node
+from the `CephCluster` resource, OSD removal will **not** alter the `CephCluster` resource's
+`storage` specification with the exception of Drive Groups when needed. While modifying these items
+might be convenient for users, there are a lot of opportunities in development to miss corner cases
+or introduce logic bugs, which we should avoid.
+
+Users should be advised in documentation that if they plan to use the Ceph CLI or Dashboard to
+manage OSD addition in the Rook-Ceph cluster, it is recommended to set `useAllDevices: false`, leave
+both disk pattern matching empty, and to not specify any other disks in the `CephCluster` resource
+to avoid giving Rook the ability to re-add OSDs when the user does not wish it to. We should also
+note to users that Rook can automatically re-add removed OSDs if they are zapped manually by the
+user via documentation to avoid possible confusion and annoyance.
+
+#### Zapping OSDs
+This design proposes to add a section to the `CephCluster` resource's `storage` configuration
+section to allow OSDs to be zapped by Rook. Note that zapping is not the same as removing an OSD.
+Users are still expected to use either manual steps or the Ceph Orchestrator interfaces to remove
+OSDs from Ceph.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+spec:
+  storage: # preexisting
+    zapOSDs: # proposed new
+      - FSID: < string, FSID of OSD to zap > # proposed new
+        controlledByCephOrchestrator: < bool, default false >
+```
+
+Zapping disks requires running a `ceph-volume` command to remove the disk. Ceph's Rook Orchestrator
+Mgr Module can modify the `CephCluster` resource to identify a specific OSD which Rook should zap
+using `ceph-volume` by adding the OSD to the `storage:zapOSDs` spec.
```

Comment:
Do we need to add CRUD functionality for zapping OSD in the orchestrator-cli and in the dashboard?